### PR TITLE
Feature/add keywords to new searchbar

### PR DESCRIPTION
**Pull request by william-jensen01**
Divided search results into sections "students" and "teachers". Right now each name will appear in both sections as we don't have user types implemented so we can't filter based off that quite yet. Created functionality that will allow the user to type keywords such as "students" and "teachers" and get that specific section of the search results.

### CODE REVIEW

**Comment by william-jensen01** [comment]
[Add keywords to new searchbar for 'students' and 'teachers'](https://trello.com/c/WeHQEMho/45-add-keywords-to-new-searchbar-for-students-and-teachers)

File: src/components/search/SearchField.js [hunk 1]
```diff
@@ -12,14 +12,29 @@ const SearchField = ({ value, toggle, setToggle, fieldRef, createModal }) => {
   //state variables
   const [options, setOptions] = useState([]);
   const [list, setList] = useState([]);
+  const [filtering, setFiltering] = useState(true);
+
+  // sections of the dropdown
+  const listSections = [
```
Comment (Brodt7258) [inline src/components/search/SearchField.js:18]: let's either find a way to lift this, or get it wrapped in useMemo.

File: src/components/search/SearchField.js [hunk 2]
```diff
@@ -78,16 +117,25 @@ const SearchField = ({ value, toggle, setToggle, fieldRef, createModal }) => {
...
-                  key={e.id}
-                  className="searchBar"
-                  onClick={() => createModal(<StudentInfoModal user={e} />)}
-                >
-                  {' '}
-                  {`${e.first_name} ${e.last_name}`}{' '}
-                </li>
-              ))}
+              {value.name.length > 0 && list.length > 0 ? currentListSections.map((section) => (
+                <>
```
Comment (Brodt7258) [inline src/components/search/SearchField.js:121]: gotta get a key on here. easiest way is to make it into a div and use section.title

**Review by Brodt7258 — CHANGES_REQUESTED** [review]
overall looks good and seems to function as intended. left a couple notes on small things that should be cleaned up before we merge. otherwise good to go.

File: src/components/search/SearchField.js (hunk 1 above)
Comment (william-jensen01) [reply src/components/search/SearchField.js]: Thanks for catching that. I removed the filtering state and did a simple check in the filterData function. First time using useMemo but I can see why you suggested it.

File: src/components/search/SearchField.js (hunk 2 above)
Comment (william-jensen01) [reply src/components/search/SearchField.js]: How did I miss this? Easy fix

**Review by Brodt7258 — APPROVED** [review]
requested changes made, pr approved.

**Comment by danielterry2121** [comment]
Looks good to me, excellent move.

**Review by danielterry2121 — APPROVED** [review]
Looks great , excellent changes.

**Pull request merged**